File: solutions_5648941810974720_0/Python/jessethegame/getting_the_digits.py
from collections import Counter

file = "A-small-attempt0.in"

DIGITS = [
    ["Z", "ZERO", 0],
    ["W", "TWO", 2],
    ["X", "SIX", 6],
    ["G", "EIGHT", 8],
    ["U", "FOUR", 4],
    ["F", "FIVE", 5],
    ["O", "ONE", 1],
    ["H", "THREE", 3],
    ["V", "SEVEN", 7],
    ["I", "NINE", 9],
]

def getting_the_digits(word):
    digits = []
    counter = Counter(word)
    for letter, word, digit in DIGITS:
        count = counter[letter]
        for i in range(count):
            digits.append(digit)
        for letter in set(word):
            counter[letter] -= count

    return ''.join(map(str, sorted(digits)))

with open(file) as handle:
  T = int(handle.readline())

  for t in range(T):
    S = handle.readline().strip()

    print "Case #{}: {}".format(t + 1, getting_the_digits(S))

File: solutions_5648941810974720_0/java/aanastasov/GettingTheDigits.java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class GettingTheDigits {
    public static void main(String[] args) throws FileNotFoundException {
        final Scanner reader = new Scanner(new File(args[0] + ".in"));
        final PrintWriter writer = new PrintWriter(new File(args[0] + ".out"));
        final int testsCount = reader.nextInt();
        for (int test = 0; test < testsCount; ++test) {
            String input = reader.next();
            int[] counts = new int[26];
            for (int i = 0; i < input.length(); ++i) {
                int ch = input.charAt(i) - 'A';
                counts[ch]++;
            }
            List<Character> res = new ArrayList<>();
            String order = "ZWXGUSVROE";
            String digits = "0268475319";
            String[] names = new String[] {
                    "ZERO", "ONE", "TWO", "THREE", "FOUR", "FIVE", "SIX", "SEVEN", "EIGHT", "NINE" };
            for (int i = 0; i < digits.length(); ++i) {
                int letter = order.charAt(i) - 'A';
                int digit = digits.charAt(i) - '0';
                int count = counts[letter];
                for (int j = 0; j < count; ++j) {
                    res.add(digits.charAt(i));
                    for (int k = 0; k < names[digit].length(); ++k) {
                        int ch = names[digit].charAt(k) - 'A';
                        counts[ch]--;
                    }
                }
            }
            Collections.sort(res);
            String result = "";
            for (Character ch : res) {
                result += ch;
            }
            writer.write(String.format("Case #%d: %s\n", test + 1, result));
        }
        reader.close();
        writer.close();
    }
}
